[Pfem] Recovers WeakPointerVector as neighbour storage

josep-m-carbonell opened the pull request:
Serialization of vectors containing raw pointers is not possible now and must be implemented. However with raw pointer containers the serialization is broken. This pull request recovers the weak_ptr container, which is not breaking the serialization. It includes PR #3970 needed to compile correctly.

josep-m-carbonell committed:
weak ContactApp

josep-m-carbonell committed:
Weak DelaunayApp

josep-m-carbonell committed:
Weak PfemApp

josep-m-carbonell committed:
Weak PfemSolidApp

josep-m-carbonell committed:
Weak SolidApp

josep-m-carbonell committed:
Weak PfemFluidApp

josep-m-carbonell committed:
echo level comment Delaunay App

josep-m-carbonell committed:
small corrections for restart

josep-m-carbonell committed:
hotfix comment

josep-m-carbonell committed:
Merge branch 'master' into pfem/master_items_weak_ptrs

josep-m-carbonell committed:
adding stl io two missing classes

josep-m-carbonell committed:
adding stl io two missing classes

josep-m-carbonell committed:
Merge branch 'core/add_stl_io_set_and_weak_ptr' into pfem/master_items_weak_ptrs

philbucher committed:
Update kratos/utilities/stl_io.h

Co-Authored-By: josep-m-carbonell <cpuigbo@cimne.upc.edu>

philbucher committed:
Update kratos/utilities/stl_io.h

Co-Authored-By: josep-m-carbonell <cpuigbo@cimne.upc.edu>

josep-m-carbonell committed:
Changing rVec to rSet

lluis-mv committed:
typo

lluis-mv reviewed (dismissed):
Hey @josep-m-carbonell ,
It works for me! (it writes and loads correctly *rest files)
[however, now I have  a little misadjustment. When a case is reloaded, the number of equations is not correctly set in the BuilderAndSolver. I will try to take a look at it tomorrow]

josep-m-carbonell commented:
Hi @lluis-mv 
   No problem, I think this "misadjustment" can be related to "reform_dofs" flag after restart. This flag will be moved to time_integration_settings json parameters soon.

josep-m-carbonell committed:
fixing stl_io tests for set and weak_ptr

adding also a new STRING_CHECK instead of the standard check to print compared values correctly.

josep-m-carbonell committed:
Merge branch 'core/add_stlio_set_and_weakptr' into pfem/master_items_weak_ptrs
